### PR TITLE
SALTO-3022: Support IssueType to IssueTypeScheme connection changes

### PR DESCRIPTION
**Pull request by alonstern**
Added support for modifying issue types in issue type scheme in Jira dc

Plugin PR: https://github.com/salto-io/jira-dc-app/pull/43

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/product_settings/data_center/data_center.ts [hunk 1]
```diff
@@ -266,6 +266,18 @@ const PLUGIN_URL_PATTERNS: UrlPattern[] = [
     httpMethods: ['put'],
     url: '/rest/api/3/resolution/.*',
   },
+  {
+    httpMethods: ['put'],
+    url: '/rest/api/3/issuetypescheme/.*/issuetype',
```
Comment (IdoZakk) [inline packages/jira-adapter/src/product_settings/data_center/data_center.ts:271]: the issue type scheme is a scheme and must contain at least one digit, use //d and + instead

Comment (IdoZakk) [reply packages/jira-adapter/src/product_settings/data_center/data_center.ts:271]: Also relevant for the other added APIs

**Review by IdoZakk — COMMENTED** [review]
Looks good 👀

File: packages/jira-adapter/src/product_settings/data_center/data_center.ts (hunk 1 above)
Comment (alonstern) [reply packages/jira-adapter/src/product_settings/data_center/data_center.ts:271]: The scheme does must contain only digits, but if it's not digits we still don't want to go to the wrong URL, we want to get to our endpoints and get and error that the id is invalid

**Review by IdoZakk — APPROVED** [review]
🥅

**Pull request merged**